### DISCUSSION TITLE
Error in running the second tutorial: CSC with VASP PLOs, PrNiO3

### DISCUSSION THREAD

**Discussion by YUBODRZHANG** [TRIQS/solid_dmft#58] [Q&A]
---

Dear Community Members,

I'm currently working through the second tutorial which focuses on the CSC with VASP PLOs for PrNiO3. Unfortunately, I've encountered an error that has halted the code's execution:

`AssertionError: Number of bands is inconsistent at ik = 0.`

Being new to this particular codebase, I'm somewhat at a loss as to what this issue might be. Any assistance or guidance you could provide would be greatly appreciated.

Sincerely,
Yubo

<img width="949" alt="ERROR" src="https://github.com/TRIQS/solid_dmft/assets/26913291/f81713f4-4b2c-419e-827d-a625287c4d32">

**Comment by the-hampel** [comment]
Hi @YUBODRZHANG,

this seems odd to me. Could you please specify which Vasp version and solid_dmft version you are using? 

You strictly followed the tutorial and all the input files match the ones we provided in the tutorial? Could you please attach the std out files from solid_dmft here? That would help a lot to narrow the problem down. 

@merkelm in case you have any ideas.

**Reply by YUBODRZHANG** [reply]
Dear Alex (and Maximilian),

I appreciate your swift response!

I have been utilizing **vasp.6.3.0** and have made modifications to the four Fortran files as per the instructions provided in "_solid_dmft-3.2.0/Docker/vasp_diffs/_". I have also made changes to the **.object** file in line with the online guidelines (change the orders). The most recent version, **solid_dmft_3.2.0**, is in use.

Regarding the input file, the parameter "**mu_gap_gb2_threshold = 0.05**" has been commented out. This parameter triggers **MaxEnt**, which isn't correctly set up on my HPC, thus causing the simulation to stop at the "_Running MaxEnt on block "total" now_" stage. I am currently working on making **MaxEnt** operational.

I have attached all the required io files to this message. Please do let me know if you require further information.

[dmft_config.ini.txt](https://github.com/TRIQS/solid_dmft/files/12894734/dmft_config.ini.txt)
[INCAR.txt](https://github.com/TRIQS/solid_dmft/files/12894735/INCAR.txt)
[observables_imp0.dat.txt](https://github.com/TRIQS/solid_dmft/files/12894736/observables_imp0.dat.txt)
[observables_imp1.dat.txt](https://github.com/TRIQS/solid_dmft/files/12894737/observables_imp1.dat.txt)
[OSZICAR.txt](https://github.com/TRIQS/solid_dmft/files/12894738/OSZICAR.txt)
[OUTCAR.txt](https://github.com/TRIQS/solid_dmft/files/12894740/OUTCAR.txt)
[stderr.txt](https://github.com/TRIQS/solid_dmft/files/12894741/stderr.txt)
[stdout.txt](https://github.com/TRIQS/solid_dmft/files/12894742/stdout.txt)

Many thanks,
Yubo

**Reply by the-hampel** [reply]
Hi,
your first 5 DMFT iterations look correct. The first handover to Vasp works, i.e. the first time the function `calc_density_correction` is called it works correctly. The problem occurs when the next Vasp SCF iteration is done and the projectors have to be updated. This goes wrong. This output shows that the projectors are suddenly very wrong: 
```
    Site 1
     0.3575323    -0.0667357
    -0.0667357     0.6601289
      trace:  1.0176612046088351
    Site 2
     1.7334907    -0.0093745
    -0.0093745     1.7625605
      trace:  3.496051209755486
    Site 3
     1.7287758    -0.0135001
    -0.0135001     1.7485661
      trace:  3.4773418666821136
    Site 4
     0.3400242    -0.1115460
    -0.1115460     0.6296493
      trace:  0.969673536955671
 ```
around line 2000 in the std output. I am not sure what goes wrong here. @merkelm did you recently rerun the example? I will also rerun it the coming days to see if I can reproduce the error. Currently I cannot see any error on your side.

**Reply by YUBODRZHANG** [reply]
Hello Alex,

Looking foward to your message. Once again, my gratitude extends to both you and @merkelm !

Best,
Yubo 
Minjiang University

**Reply by the-hampel** [reply]
Hi @YUBODRZHANG , 
quick update from my side. I was able to reproduce your problem. This is due to the recent fixes we made in the CSC interface in Vasp, that now require less aggressive mixing in Vasp. I will fix the tutorial input files and update them shortly.

**Reply by the-hampel** [reply]
Hi @YUBODRZHANG ,

this is now fixed. Please see the current unstable branch (I will backport the changes once all tests passed) of solid_dmft for updated input files: 
https://github.com/TRIQS/solid_dmft/commit/0fb1c07733b0227b31f3377833eeddfb3074846d

The new tutorial is already online, and replaced the old one in the documentation:
https://triqs.github.io/solid_dmft/tutorials/PrNiO3_csc_vasp_plo_cthyb/tutorial.html

Let me know if there are still problems

**Reply by YUBODRZHANG** [reply]
Hi Alex, 

It works. Thank you!

Yubo
